Using compiler.has_function() without includes specified breaks clang-16

The upcoming clang-16 release treats implicit function declarations as fatal errors (see https://discourse.llvm.org/t/configure-script-breakage-with-the-new-werror-implicit-function-declaration/65213). As a result, using `compiler.has_function()` from distutils without explicitly specifying necessary includes results in checks failing incorrectly, e.g.:

```
python3.10 setup.py build -j 1 -j1
/tmp/portage/dev-python/pytables-3.8.0/temp/blosc2_list_compressors2avhs2_r.c:2:5: error: call to undeclared function 'blosc2_list_compressors'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    blosc2_list_compressors();
    ^
1 error generated.
[...]
* Could not find blosc2 headers and library; disabling support for it.
```

(this is with blosc2 library being available)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -45,3 +45,39 @@ jobs:
         ptrepack -h
         ptdump -h
         pttree -h
+
+
+  build_cblosc:
+    name: Sdist with cblosc2
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        submodules: true
+    - name: Set up Python
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        auto-update-conda: true
+        channels: conda-forge
+        channel-priority: strict
+    - name: Install dependencies
+      run: |
+        conda install setuptools pip wheel build packaging py-cpuinfo numpy cython numexpr bzip2 hdf5 lzo 'typing_extensions<4.2' c-blosc2
+    - name: Source distribution
+      run: |
+        python -m build --sdist
+    - name: Installation
+      run: |
+        pip install -v dist/*.tar.gz --no-deps
+    - name: 'Run test'
+      run: |
+        cd .. && python -m tables.tests.test_all -v
+        pt2to3 -h
+        ptrepack -h
+        ptdump -h
+        pttree -h

--- a/setup.py
+++ b/setup.py
@@ -262,17 +262,24 @@ if __name__ == "__main__":
 
     debug = "--debug" in sys.argv
 
-    blosc2_inc, blosc2_lib = get_blosc2_directories()
-
     # Global variables
-    lib_dirs = [blosc2_lib]
-    inc_dirs = [Path("hdf5-blosc/src"), Path("hdf5-blosc2/src"), blosc2_inc]
+    lib_dirs = []
+    inc_dirs = [Path("hdf5-blosc/src"), Path("hdf5-blosc2/src")]
     optional_libs = []
     copy_libs = []
 
     default_header_dirs = None
     default_library_dirs = None
     default_runtime_dirs = None
+
+    blosc2_inc = blosc2_lib = None
+    if os.environ.get("PYTABLES_NO_BLOASC2_WHEEL", None) is None:
+        try:
+            blosc2_inc, blosc2_lib = get_blosc2_directories()
+            lib_dirs.append(blosc2_lib)
+            inc_dirs.append(blosc2_inc)
+        except EnvironmentError:
+            pass
 
     if os.name == "posix":
         prefixes = ("/usr/local", "/sw", "/opt", "/opt/local", "/usr", "/")
@@ -691,7 +698,7 @@ if __name__ == "__main__":
     # Try to locate the compulsory and optional libraries.
     lzo2_enabled = False
     compiler = new_compiler()
-    if not BLOSC2_DIR:
+    if not BLOSC2_DIR and blosc2_inc is not None:
         # Inject the blosc2 directory as detected
         BLOSC2_DIR = os.path.dirname(blosc2_inc)
     for (package, location) in [

--- a/setup.py
+++ b/setup.py
@@ -273,7 +273,7 @@ if __name__ == "__main__":
     default_runtime_dirs = None
 
     blosc2_inc = blosc2_lib = None
-    if os.environ.get("PYTABLES_NO_BLOASC2_WHEEL", None) is None:
+    if os.environ.get("PYTABLES_NO_BLOSC2_WHEEL", None) is None:
         try:
             blosc2_inc, blosc2_lib = get_blosc2_directories()
             lib_dirs.append(blosc2_lib)

